temporalio/worker/_workflow_instance.py

```diff
@@ -789,44 +791,51 @@ def workflow_patch(self, id: str, *, deprecated: bool) -> bool:
             command.set_patch_marker.deprecated = deprecated
         return use_patch
 
+    def workflow_payload_converter(self) -> temporalio.converter.PayloadConverter:
+        return self._payload_converter
+
     def workflow_random(self) -> random.Random:
         return self._random
 
     def workflow_set_query_handler(
         self, name: Optional[str], handler: Optional[Callable]
     ) -> None:
         if handler:
-            self._queries[name] = temporalio.workflow._QueryDefinition(
+            defn = temporalio.workflow._QueryDefinition(
                 name=name, fn=handler, is_method=False
```

Comment:
IMO for this & signal it'd be better to use some special sigil type to represent the dynamic handler rather than just `None`.

Reply:
We have used `None` to represent dynamic for signals/queries for a long time now. Not sure it's worth changing.